### ISSUE TITLE
Should dynamic fields retain values on work item update?

### ISSUE THREAD

**Issue by rgarg1** [openshiftio/openshift.io#3951]
### A question for the UX team....

Consider the following scenario:

1. User open work item quick preview
2. Enters some value on a dynamic field like _Story points_. Doesn't saves the value yet
3. Edits the iteration of the work item. This causes the work item to reload and the storypoints entered earlier by the user are lost.

So, is it okay to lose data in the dynamic fields on an update, or the data must be retained is the question.

**Comment by rgarg1** [comment]
Cc @Veethika @divyanshiGupta

**Comment by kwk** [comment]
@rgarg1 

> So, is it okay to lose data in the dynamic fields on an update, or the data must be retained is the question.

It is absolutely NOT okay to discard the entered data!

**Comment by Veethika** [comment]
@rgarg1 if the user has edited any field in the quick preview panel and has not committed to it or saved it, the moment they perform an action that calls for the closing of the panel, a warning should appear asking them to commit to the change or discard it.

**Comment by muruGanesan** [comment]
@rgarg1, I believe this issue is closed or addressed from the UX side. If yes, you can remove UX labels.

@Veethika, @AshishDurgude, @sunilmalagi, @sdash-redhat, would you link to capture this UX recommendation as part of UX standards in our design wiki?

**Comment by rgarg1** [comment]
Done @muruGanesan

**Comment by muruGanesan** [comment]
thanks @rgarg1

**Comment by kwk** [comment]
I consider this a high priority because of Data loss.

**Comment by nimishamukherjee** [comment]
> a warning should appear asking them to commit to the change or discard it.

@openshiftio/uxd-team please provide a visual of the warning.
cc @Veethika @sunilmalagi

**Comment by joshuawilson** [comment]
I don't think we need a warning as much as auto-save. When you leave the field it should save the information.

**Comment by rgarg1** [comment]
+1 to @joshuawilson

**Comment by nimishamukherjee** [comment]
> I don't think we need a warning as much as auto-save. When you leave the field it should save the information.

@Veethika @serenamarie125 please comment ^^

**Comment by divyanshiGupta** [comment]
@Veethika is the UX ready for this?

**Comment by joshuawilson** [comment]
@divyanshiGupta for now, you can just auto-save it. That is the normal use of a form.

**Comment by divyanshiGupta** [comment]
@joshuawilson okay :+1:

**Comment by christianvogt** [comment]
@Veethika The UX here has to change if we implement auto save. We cannot have an ok cancel button if we immediately apply the change on leaving the field.

**Comment by Veethika** [comment]
> Btw I think the original issue was that editing a separate field causes data to lose in an unsaved field. This is indeed a bug. The user should have the ability to go back to the first field and save or discard the value. Instead, the value was thrown away unexpectedly. It had nothing to do with closing the panel.

@christianvogt Right, my solution swayed towards the opening and closing the panel which should be addressed in a different story altogether. For this issue, the work-item should not be reloaded everytime a user begins to edit a field. 
But the state of the fields has to be maintained for a 'session' if I'm not wrong. So are we referring to end of the session as 'until another work item is selected'(the content of quick preview changes)?

**Comment by christianvogt** [comment]
@Veethika @joshuawilson 
I believe this issue is exactly the same as I logged here: https://github.com/openshiftio/openshift.io/issues/4643

What we do if the user chooses to navigate away is a separate issue. But when editing the same issue, we are losing data. If we fix this problem, do we still need an auto save? Or a warning about navigating away?

**Comment by joshuawilson** [comment]
@christianvogt can you close #4643 as a duplicate?

**Comment by divyanshiGupta** [comment]
> What we do if the user chooses to navigate away is a separate issue. But when editing the same issue, we are losing data. If we fix this problem, do we still need an auto save? Or a warning about navigating away?

@Veethika what is the suggested UX?

**Comment by divyanshiGupta** [comment]
Unassigning myself as I am not working on this issue currently as I have some other tasks on priority.
